src/lib/rng/jitter_rng/jitter.cpp

```diff
@@ -0,0 +1,64 @@
+/*
+* jitterentropy utility functions
+* (C) 2024 Planck Security S.A.
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#include <span>
+
+#include <botan/exceptn.h>
+#include <botan/internal/jitter.h>
+
+#include <memory>
+
+extern "C" {
+#include <jitterentropy.h>
+}
+
+namespace Botan {
+
+rand_data* jitter_collector_create() {
+   static int result = jent_entropy_init();
+
+   if(result != 0) {
+      throw Internal_Error("jitter entropy can not be used");
+   }
+
+   unsigned int oversampling_rate = 0;  // use default oversampling
+   unsigned int flags = 0;
+
+   rand_data* jitter = jent_entropy_collector_alloc(oversampling_rate, flags);
+   if(!jitter) {
+      throw Internal_Error("jitter entropy could not be allocated");
+   }
+
+   return jitter;
+}
+
+void jitter_collector_free(rand_data* collector) {
+   if(collector) {
+      jent_entropy_collector_free(collector);
+   }
+}
+
+void jitter_collect_into_buffer(rand_data* collector, std::span<uint8_t> buf) {
+   jitter_collect_into_buffer(collector, buf.data(), buf.size());
+}
+
+void jitter_collect_into_buffer(rand_data* collector, uint8_t buf[], size_t len) {
+   if(!collector) {
+      throw Invalid_State("no jitter entropy collector has been allocated");
+   }
+
+   if(len <= 0) {
```

Comment:
An RNG should accept (as a no-op) an output length of zero; output 0 empty bytes always succeeds no matter what the RNG state!